### DISCUSSION TITLE
Navigation with route obj keeps previous params

### DISCUSSION THREAD

**Discussion by oswaldofreitas** [vuejs/router#2300] [Help and Questions] [answered]
### Reproduction

https://play.vuejs.org/#eNqdVstu20YU/ZVrZiEKkUm/kgUrq25dA0nhOIZbtCjKLMbkSGI8nCFmhpIMR8vu+gntz/VLeufBh2XJDrIQKN7Hua8zd/gQlKTg0WcVJEFRVkJqeIBMUqLpD1UFa5hKUcJgUdNByr2BFLWm0mui2L12auPX6EhVRc435S1qiL9hygGiWtHQubv3UtRch4NX6DYYBqPAu2NuY03LiqH/xBiOK/vAP0pLwWeT81pKyn1qUBE9T8ax18HDg5NH05qxa9TBem1RYgcz5mTR4N3YbC4LfgdanKZBnAaTd6Kk47jT7LStpMjrTCv0ufZ/IfxdyDs1fME/sQAPwElJExh4b7gslB7A+hHeT4IqPtCwRNyXYF1alLeZxRmRiHZO5BPPcdy0YfyeY+vQvBBcQexkhiaPI/xW0GWjjb16HPfmhK8qk0WlQVFdm163DMPBW5A+wfY3ifQojYZRfWFHLXxphn/agodIKmSBTWGCbHL4W7n+gZZC3r/DduNj5IWuzO0ptklekxltkzMv25JSmNWfplHthA2nBiPHVBjECSN89j0KMoGwHKmcWGQUzAuWI7cTB4AQ7gFbuTJqlB64mXun2AgAa6/ZAbsTMU4Uq2c7cUdQUk2SDnZO1CXWmABOjzrh2gf/ZB6fNlomsWX9MYQWae4mlGwbWzi0aK7h+HeN40+5H/otnQpJL0g2D0MtRnZiI+B0pYdwiivCeJrogtGIiVk46BxwKlpEFZGkVMPvDCZAMYVwr5VGZnrw5YuxM2VHvthhU75E/ktuw4UGAmt3OK5gTpfuNJwiQaIo0gLWPlBOGUVFY9EP6C0sqHVrjEYYD89ghmM0zYa1Cem6QVeWtDmdkpo1ncaj0RD36aadH03a/WmogbtzHKPQKrdsy2YjVJJ2ji5r62rkX78nugCjF3bG1y4BB71fkgoXgeBYsJ1R6hW4u1vSpkEXwYjTYK51pZI4rnl1N4uQ8nFncXYSHUVv4hzJ2JNGVJX7t1Isld08aeAPTBqcoVGc04UWgql9UhW7QjwxPHsbvY0OY1bcxogeFzynK4vtibXGMrXCdkyL2UaR5pQWjMqPlV2gj4oljInlz1ZmaNMmms1pdrdF/lmtXMrXkmJ1C9orThM5o9qpL365Qor2lCVuEIbWzyhvKJ7E2uTozH6seY5p9+xstu/tzAo++1VdrDTlqimqWTJuwdhBnj9TepfucXTS66LS94yqKFPmysDrEbca3nPO71bInMoEjqoVYLJFDq8ODg7s4S4RruD7t0JrUSZweFCtrLwieY7JthK7BBAWJkDgNf4ssPdmdIq7tLHEZDZvvi1n9XjSN8JzeuwVJ5OOkmCJisoTpxSsuddZMTlnRYaHeU6xK/iVgOeZ4eEbI9m2Gk0LiWfOUKD58tlm/4eoB4zh6abWqflo699c3dfaAj8rdsbrJwVkhsPYML0SS8isueDW44jn0H48bcntv3/+gqsio3twL2rICP+WNMexb+KTTh9HB693tXqOS9yF6rksqDQsxn1qvAHZ1s/FER4+kKrLRJPbjZpuaIW3IyhNK//hdGjw3gDBZtjjbEH9fQd432025d+/4aOo1B7Gnxa8wI3KhMCYj+t9dou3Szflwfp/DBsugg==

## vue-router 4.2.5
1. Click the "Car" link
2. Click the first "Products" link
3. You'll see the Product List view
4. Click the "Car" link again
5. Now click on the 2nd Products link
6. ✅ Nice! you can see the Product List view

##vue-router 4.3.0+
1. Change the vue-router version to 4.3 in the Import Map tab
2. Repeat steps from 1 to 5 and check the console log
3. ❌ Oops! infinite loop

### Expected behavior

When navigating with a raw obj like `:to="{ name: 'Product List' }"` should have the same result as `to="/products"` string

### Actual behavior

It's entering in an infinite loop

### Additional information

I've put a console log in router.js > `beforeEach()` so that you can check that the `lang` param isn't being removed in the newer version of vue-router

**Comment by posva** [comment, marked as answer]
This was a fix in #2031 
You can pass the [optional param as an empty string instead](https://play.vuejs.org/#eNqdVstu20YU/ZVrZSEKkUm/kgUrq25dA0nhOIZbtCjKLEbkiGJMzgxmhrYMRcvu+gntz/VLeufBh2XJDrIQKN7Hmfs49w5Xg4oULPysBvGgqASXGlaQSko0/UEIWMNc8gqGdzUdJswbSF5rKr0mjNxrpzZ+jY4IETrfhLWoAf5GCQMIa0UD5+7eK14zHQxfodtwNBgPvDvGNtG0EiX6T43hRNgH/lFacpZPz2spKfOhgSB6EU8ir4PVysnDeV2W16iD9dqiRA5mwshdg3djo7ks2C1ofpoMomQwfccrOok6zU5bIXlWp1qhz7X/C8HvXN6q0Qv+sQVYASMVjWHoveGyUHoI60d4P3Gq2FDDPeK+BOvCoqyNLEqJRLRzIp94TqKmDJP3DEuH5gVnCiInMzR5fMJvBb1vtJFXT6Jen/BVpbIQGhTVtal1yzBsvAXpE2x/k0iPwmgY1Rd21MKXpvmnLXiApEIW2BCmyCaHv5XrH2jF5cM7LDc+xl7o0tweYhvkNclpG5x52RaUwqj+NIVqO2w4NRw7psIwikvC8u9RkHKEZUjl2CKjYFGUGXI7dgAI4R6wlSvjRumBm753io0DYO01O2B3IkaxKut8J+4YKqpJ3MEuiLrEHGPA7lEnXPvDP5nHp42SSSxZvw2BRVq4DsXb2haMLJorOP5dY/sT5ps+o3Mu6QVJF0Gg+dh2bASnuByMjzmXlzQseR4MO1Psh+ahIJJUavSdQQMo5hDstdLQ9A2+fDF2JuHQpzlqEpfIfMnapHHvmEdGZ3WeU+nevE2wgjAMTXAOG4vnBP6sMZjDsAW4EcbohGOWYqcMNKxNeC5hurS8zOic1GVTTGR/w82ny3RxNG1XpOk+rsdJhEKr3LIQm6EXknaOLkbrauRfvwq6A8YvrIWvnXMHvV8RgbPOGSZsm5F4Ba7nlpfJoDvBiJPBQmuh4iiqmbjNQ2R11FmcnUQZcq0nCamq9meS3yu7WJKBn4dkcIZGUUbvNOel2iei2AX/xPDsbfg2PIzKYhYhelSwjC4ttifQGlPUCksxL/KNBM0QFiWVH4Xdj48SJWXJ73+2MkOZNtB0QdPbLfLPaulCvpYUs7ujveQ0kTnVTn3xyxVd4v9WWeGCKNH6GeUNxXGrTYzO7MeaZRh2z85G+972q2D5r+piqSlTTVLtOFl728TzZ1Lvwj0OT3pVVPqhpCpMlbkR8PbDpYXXmPObcZlRGcORWAIGW2Tw6uDgAKcM0EjmBdufca15FcPhgVhauSBZhsG2EjvsCAtTIPAafxbYe5d0jquyscRgNi+2LXN6PO0b4Ywee8XJtKMknIRH4RtUnjglL5truyym52WR4iAvKFYFPwJwlkscvAmSbavRvJA4b4YCzYfNNvs/eD0sS5xsap2ab7L+xdR9jN3hV8PO8/pBAcmxGRumV/weUmvOcamixxHLoP022hLbf//8BVdFSvfggdeQEvYtYU4iX8QnlT4OD17vKvUCt7U7qudyR6VhMe5S4w3Itn4sjvDwgYguEk1mGzndUIGXHyhNhf8uOjR4b4BgMew4W1B/qQFeaptF+fdv+MiF2sPz5wUrcJuWnOOZj/N9doO3Czdhg/X/NH4ilg==). BTW you can also not use `next` (better) and copy just the params and reuse the name instead of copying the whole to object

**Comment by oswaldofreitas** [comment]
nice, setting the param with empty fixed and not having to call `next()` anymore is a great improvement, thanks!
